torch/onnx/symbolic_opset11.py

```diff
@@ -830,3 +830,100 @@ def prim_ConstantChunk(g, self, chunks, dim):
         res.append(g.op("Slice", self, start, end, axis))
         start = end
     return res
+
+def repeat_interleave(g, self, repeats, dim=None):
+    from torch.onnx.symbolic_opset9 import reshape
+    input = self
+    final_dim = dim
+    # if dim is None flatten
+    # By default, use the flattened input array, and return a flat output array
+    if sym_help._is_none(dim):
+        input = reshape(g, self, g.op("Constant", value_t=torch.tensor([-1])))
+        dim = 0
+    else:
+        dim = sym_help._maybe_get_scalar(dim)
+
+    repeats_dim = sym_help._get_tensor_rank(repeats)
+    repeats_sizes = sym_help._get_tensor_sizes(repeats)
+    input_sizes = sym_help._get_tensor_sizes(input)
+    if repeats_dim is None:
+        raise RuntimeError('Unsupported: ONNX export of repeat_interleave for unknown '
+                           'repeats rank.')
+    if repeats_sizes is None:
+        raise RuntimeError('Unsupported: ONNX export of repeat_interleave for unknown '
+                           'repeats size.')
+    if input_sizes is None:
+        raise RuntimeError('Unsupported: ONNX export of repeat_interleave for unknown '
+                           'input size.')
+
+    input_sizes_temp = input_sizes.copy()
+    perm_i = [0]
+    for idx, input_size in enumerate(input_sizes):
+        perm_i.append(idx + 1)
+        if input_size is None:
+            input_sizes[idx], input_sizes_temp[idx] = 0, -1
+    perm_i[0], perm_i[dim] = perm_i[dim], perm_i[0]
+
+    # Cases when repeats is a single value tensor and dim has unknown input size
+    if (repeats_dim == 0 or (repeats_dim == 1 and repeats_sizes[0] == 1)) and input_sizes[dim] == 0:
+        if not sym_help._is_tensor(repeats):
+            repeats = g.op("Constant", value_t=torch.LongTensor(repeats))
+        reps = sym_help._size_helper(g, input, dim)
+        reps = unsqueeze(g, reps, 0)
+        repeats = g.op("Expand", repeats, reps)
+    elif repeats_dim == 1 and input_sizes[dim] == 0:
```

Comment:
is `repeats_dim == 1` redundant?

Reply:
Both are different cases as 
1) [0] ,single value 1-dim tensor
2) [0, 1, 2] ,multi value 1-dim tensor